qemu/tests/hv_vapic_test.py

```diff
@@ -0,0 +1,130 @@
+    """
+    Test the hv_vapic flag improvement
+    1) Create tmpfs data disk in host
+    2) Mount&format the disk in guest, then prepare the fio test
+       environment
+    3) Boot the guest with all hv flags
+    4) Run fio test, record the result's bw value
+    5) Shutdown and boot the guest again without hv_vapic flag
+    6) Run fio test again, record the result's bw value
+    7) Calculate the improvement value of the 2 fio tests,
+       then check if it is obvious enough
+
+    param test: the test object
+    param params: the test params
+    param env: the test env object
+    """
+
+    def _create_tmpfs_data_disk():
+        """
+        Create a tmpfs data disk
+        """
+        logging.info("Create tmpfs data disk")
+        disk_name_key = params["disk_name_key"]
+        tmp_dir = data_dir.get_tmp_dir()
+        tmpfs_dir = os.path.join(tmp_dir, "tmpfs")
+        process.system("mkdir -p " + tmpfs_dir, shell=True)
+        params[disk_name_key] = os.path.join(tmpfs_dir, "data")
+        tmpfs_image_name = params["tmpfs_image_name"]
+        img_param = params.object_params(tmpfs_image_name)
+        img = QemuImg(img_param, data_dir.get_data_dir(), tmpfs_image_name)
+        img.create(img_param)
+
+    def _format_tmpfs_disk():
+        """
+        Format the new tmpfs disk in guest
+
+        return: the formatted drive letter of the disk
+        """
+        logging.info("Boot the guest to setup tmpfs disk")
+        vm, session = _boot_guest_with_cpu_flag(cpu_model_flags)
+        logging.info("Format tmpfs disk")
+        disk_size = params["image_size_" + params["tmpfs_image_name"]]
+        disk_id = utils_disk.get_windows_disks_index(session, disk_size)[0]
+        drive_letter = utils_disk.configure_empty_windows_disk(
+                    session, disk_id, disk_size)[0]
+        vm.graceful_shutdown(timeout=timeout)
+        return drive_letter
+
+    def _boot_guest_with_cpu_flag(hv_flag):
+        """
+        Boot the guest, with param cpu_model_flags set to hv_flag
+
+        param hv_flag: the hv flags to set to cpu
+
+        return: the booted vm and a loggined session
+        """
+        params["cpu_model_flags"] = hv_flag
+        params["start_vm"] = "yes"
+        vm_name = params["main_vm"]
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        session = vm.wait_for_login(timeout=timeout)
+        return (vm, session)
+
+    def _run_fio(session, drive_letter):
+        """
+        First format tmpfs disk to wipe out cache,
+        then run fio test, and return the result's bw value
+
+        param session: a session loggined to guest os
+        drive_letter: the drive to run the fio job on
+        return: the bw value of the running result(bw=xxxB/s)
+        """
+        bw_search_reg = params["bw_search_reg"]
+        logging.info("Format tmpfs data disk")
+        session.cmd(params["format_disk_cmd"] % drive_letter, timeout=timeout)
```

Comment:
Why format data disk again? the `configure_empty_windows_disk` has done it.

Reply:
The format is to clear the disk cache of the previous fio test.

Reply:
Got it, thanks for your explanations.

Reply:
So could we use `create_filesystem_windows(session, partition_name, fstype, timeout=360, quick_format=True)` from utils_disk.py to replace it ?